### PR TITLE
Feature/666 improve loganalytics logging

### PR DESCRIPTION
**Pull request by zzeekk**
### What changes are included in the pull request?
see #666

### CODE REVIEW

File: sdl-core/src/main/scala/io/smartdatalake/util/misc/SmartDataLakeLogger.scala [hunk 1]
```diff
@@ -38,9 +39,11 @@ trait SmartDataLakeLogger {
     e
   }
 
-  private[smartdatalake] def logWithSeverity(severity: Level, msg: String): Unit = {
+  private[smartdatalake] def logWithSeverity(severity: Level, msg: String, cause: Throwable): Unit = {
     severity match {
+      case Level.ERROR if Environment.includeDAGResultExceptionInLog => logger.error(msg, cause)
       case Level.ERROR => logger.error(msg)
```
Comment (pgruetter) [inline sdl-core/src/main/scala/io/smartdatalake/util/misc/SmartDataLakeLogger.scala:45]: Isn't log.error called twice if includeDAGResultExceptionInLog is true, once with cause, once without?

File: sdl-azure/src/main/scala/io/smartdatalake/util/azure/LogAnalyticsAppender.scala [hunk 2]
```diff
@@ -81,13 +91,20 @@ class LogAnalyticsAppender(name: String, var backend: Option[LogAnalyticsBackend
...
   override def stop(): Unit = {
     flush()
     super.stop()
-    println("stopped")
+    println("LogAnalyticsAppender stopped")
+  }
+
+  private def enrichSparkLocalProperties(event: LogEvent) = {
+    // If we are on an executor, we have a TaskContext
+    // On executors we need to enricht LogEvent with SDLB informations from spark local properties
```
Comment (pgruetter) [inline sdl-azure/src/main/scala/io/smartdatalake/util/azure/LogAnalyticsAppender.scala]:
```suggestion
    // On executors we need to enrich LogEvent with SDLB information from spark local properties
```

**Review by pgruetter — APPROVED** [review]
Got two small comments, please check.

File: sdl-core/src/main/scala/io/smartdatalake/util/misc/SmartDataLakeLogger.scala (hunk 1 above)
Comment (zzeekk) [reply sdl-core/src/main/scala/io/smartdatalake/util/misc/SmartDataLakeLogger.scala:45]: No, pattern matching always executes only one case statement and then stops.

**Pull request merged**